Step04/app/src/main/res/layout/listviewitem_weather.xml

```diff
@@ -0,0 +1,105 @@
+            <TableLayout
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="1">
+                <TableRow>
+                    <TextView
+                        android:id="@+id/textAverageTemperature"
+                        android:layout_width="60dp"
+                        android:layout_height="50dp"
+                        android:textColor="#000000"
+                        android:textSize="40dp"
+                        android:textStyle="bold"
+                        android:gravity="center_horizontal"
+                        android:layout_marginTop="10dp"
+                        android:layout_marginBottom="10dp"
+                        android:layout_weight="1" />
+                </TableRow>
+                <TableRow>
+                    <TextView
+                        android:id="@+id/textMinTemperature"
+                        android:layout_width="0dp"
+                        android:layout_height="20dp"
+                        android:textColor="#0000FF"
+                        android:gravity="right"
+                        android:layout_marginRight="5dp"
+                        android:layout_weight="1" />
+                    <TextView
+                        android:id="@+id/textMaxTemperature"
+                        android:layout_width="0dp"
+                        android:layout_height="20dp"
+                        android:textColor="#FF0000"
+                        android:gravity="left"
+                        android:layout_marginLeft="5dp"
+                        android:layout_weight="1" />
+                </TableRow>
+            </TableLayout>
+            <LinearLayout
+                android:layout_width="0dp"
+                android:layout_height="wrap_content"
+                android:layout_weight="3">
+                <ImageView
+                    android:id="@+id/imageWeather"
+                    android:layout_width="60dp"
+                    android:layout_height="60dp"
+                    android:paddingTop="10dp"
+                    android:paddingBottom="10dp"
+                    android:gravity="left"
+                    android:layout_marginLeft="5dp"
+                    android:layout_marginRight="5dp"
+                    android:layout_marginTop="10dp"
+                    android:layout_marginBottom="10dp" />
+                <TextView
+                    android:id="@+id/textDatetime"
+                    android:layout_width="200dp"
+                    android:layout_height="60dp"
+                    android:textColor="#000000"
+                    android:gravity="left|center_vertical"
+                    android:layout_marginLeft="10dp"
+                    android:layout_marginTop="10dp"
+                    android:layout_marginBottom="10dp" />
+            </LinearLayout>
+        </LinearLayout>
+    </TableRow>
+</TableLayout>
```

Comment:
@kwoolytech it's good to practice using new layout, "TableLayout". If you can achieve same UI using ```RelativeLayout``` , i recommend you to use it because you can reduce the no of depth in view hierarchy next time.